### PR TITLE
Adding external c-tor for Validation Options.

### PR DESCRIPTION
**Pull request by Mercurrent**

### CODE REVIEW

File: trillian/ctfe/handlers.go [hunk 1]
```diff
@@ -219,6 +219,25 @@ type CertValidationOpts struct {
...
+// NewCertValidationOpts can be used for struct creation outside of ctfe package.
+func NewCertValidationOpts(trustedRoots *PEMCertPool, currentTime time.Time, rejectExpired bool, rejectUnexpired bool, notAfterStart *time.Time, notAfterLimit *time.Time, acceptOnlyCA bool, extKeyUsages []x509.ExtKeyUsage) CertValidationOpts {
+	var vOpts CertValidationOpts
+	vOpts.trustedRoots = trustedRoots
+	vOpts.currentTime = currentTime
+	vOpts.rejectExpired = rejectExpired
+	vOpts.rejectUnexpired = rejectUnexpired
+	vOpts.notAfterStart = notAfterStart
+	vOpts.notAfterLimit = notAfterLimit
+	vOpts.acceptOnlyCA = acceptOnlyCA
```
Comment (daviddrysdale) [inline trillian/ctfe/handlers.go]: (style) probably neater to initialize inline:
```golang
	vOpts := CertValidationOpts{
		trustedRoots: trustedRoots,
		currentTime: currentTime,
		rejectExpired: rejectExpired,
		rejectUnexpired: rejectUnexpired,
		notAfterStart: notAfterStart,
		notAfterLimit: notAfterLimit,
		acceptOnlyCA: acceptOnlyCA,
	}
```

File: trillian/ctfe/handlers.go [hunk 2]
```diff
@@ -219,6 +219,25 @@ type CertValidationOpts struct {
 	extKeyUsages []x509.ExtKeyUsage
 }
 
+// NewCertValidationOpts can be used for struct creation outside of ctfe package.
```
Comment (daviddrysdale) [inline trillian/ctfe/handlers.go]: (nit) Comment describes how function is used, not what it does.  Maybe just "NewCertValidationOpts builds validation options based on parameters"?

File: trillian/ctfe/handlers.go (hunk 1 above)
Comment (Mercurrent) [reply trillian/ctfe/handlers.go]: Done.

File: trillian/ctfe/handlers.go (hunk 2 above)
Comment (Mercurrent) [reply trillian/ctfe/handlers.go]: Done.

**Pull request merged**